### PR TITLE
Feat/verify

### PR DESCRIPTION
**Pull request by Muncherkin**
Added verification logic to user using a VerifyInfo object for future expansion.
Added connection to verify microservice.
Added automatic verification if user has a post.

!! - - This update is dependent on the same feature in Efterphest and Auth. - - !!

### CODE REVIEW

File: src/api/user.api.ts [hunk 1]
```diff
@@ -654,4 +658,49 @@ export class UserAPI {
...
+      data: { ssn: ssn.length == 10 ? ssn : ssn.slice(2), alreadyVerified: userVerified }, //Case for both 10 and 12 digit ssn.
+    };
+    const data = await axios.request(options);
+
+    if (data.status !== 200) {
+      logger.debug('Could not verify user with given ssn');
+      throw new ServerError(
+        'Kunde inte verifiera användaren med givet personnumret. Tillhör personnumret en E:are?',
+      );
+    }
```
Comment (afroborg) [inline src/api/user.api.ts:693]: I feel like this should be extracted into a service to keep in line with how we're already doing services for other external api:s

File: src/resolvers/user.resolver.ts [hunk 2]
```diff
@@ -112,6 +114,19 @@ const userResolver: Resolvers = {
...
     },
+    verified: async ({ username }, _, ctx) => {
+      await hasAuthenticated(ctx);
+
+      if (!username) {
+        return false;
+      }
+
+      const verified = await api.isUserVerified(username);
+      const posts = await postApi.getPostsForUser(username, false);
```
Comment (afroborg) [inline src/resolvers/user.resolver.ts:125]: **Suggestion**
use Promise.all to fetch in parallell

**Review by afroborg — CHANGES_REQUESTED** [review]
add some tests for your new api functions pls

File: src/api/user.api.ts (hunk 1 above)
Comment (Muncherkin) [reply src/api/user.api.ts:693]: True!

**Comment by Muncherkin** [comment]
Made changes after review!

**Review by pontussjostedt — APPROVED** [review]
This has been reviewed before :)

**Comment by pontussjostedt** [comment]
LGTM! Reqest solved :)

**Pull request merged**